apps/admin/backend/src/tabulation.test.ts

```diff
@@ -0,0 +1,958 @@
+        votingMethod: 'absentee',
+        ...mockZooCouncilMammalSummary,
+      }),
+    ];
+
+    const groupedWriteInSummaries = tabulateWriteInTallies({
+      election,
+      writeInTallies,
+      groupBy: {
+        groupByPrecinct: true,
+        groupByVotingMethod: true,
+      },
+    });
+
+    const electionWriteInSummaries = Object.values(groupedWriteInSummaries);
+    expect(electionWriteInSummaries).toHaveLength(4);
+    for (const electionWriteInSummary of electionWriteInSummaries) {
+      expect(electionWriteInSummary.contestWriteInSummaries).toEqual({
+        'zoo-council-mammal': mockZooCouncilMammalSummary,
+        'aquarium-council-fish': getEmptyContestWriteInSummary(
+          'aquarium-council-fish'
+        ),
+      });
+    }
+    expect(electionWriteInSummaries).toMatchObject(
+      expect.arrayContaining([
+        expect.objectContaining({
+          precinctId: 'precinct-1',
+          votingMethod: 'precinct',
+        }),
+        expect.objectContaining({
+          precinctId: 'precinct-2',
+          votingMethod: 'precinct',
+        }),
+        expect.objectContaining({
+          precinctId: 'precinct-1',
+          votingMethod: 'absentee',
+        }),
+        expect.objectContaining({
+          precinctId: 'precinct-2',
+          votingMethod: 'absentee',
+        }),
+      ])
+    );
+  });
+});
+
+test('modifyElectionResultsWithWriteInSummary', () => {
+  const { election } = electionMinimalExhaustiveSampleDefinition;
+  const electionResults = getEmptyElectionResults(election);
+
+  electionResults.cardCounts.bmd = 112;
+  electionResults.contestResults['zoo-council-mammal'] = {
+    contestId: 'zoo-council-mammal',
+    ballots: 112,
+    overvotes: 6,
+    undervotes: 6,
+    contestType: 'candidate',
+    votesAllowed: 3,
+    tallies: {
+      lion: {
+        id: 'lion',
+        name: 'Lion',
+        tally: 56,
+      },
+      [writeInCandidate.id]: {
+        ...writeInCandidate,
+        tally: 56,
+      },
+    },
+  };
+
+  const modifiedElectionResults = modifyElectionResultsWithWriteInSummary(
+    electionResults,
+    {
+      contestWriteInSummaries: {
+        'zoo-council-mammal': mockZooCouncilMammalSummary,
+        'aquarium-council-fish': mockAquariumCouncilFishSummary,
+      },
+    }
+  );
+
+  expect(modifiedElectionResults.contestResults['zoo-council-mammal']).toEqual({
+    ballots: 112,
+    contestId: 'zoo-council-mammal',
+    contestType: 'candidate',
+    overvotes: 6,
+    tallies: {
+      chimera: {
+        id: 'chimera',
+        isWriteIn: true,
+        name: 'Chimera',
+        tally: 7,
+      },
+      lion: {
+        id: 'lion',
+        name: 'Lion',
+        tally: 61,
+      },
+      'write-in': {
+        id: 'write-in',
+        isWriteIn: true,
+        name: 'Write-In',
+        tally: 11,
+      },
+    },
+    undervotes: 15,
+    votesAllowed: 3,
+  });
+});
+
+describe('tabulateElectionResults', () => {
```

Comment:
Testing here is lighter than it should be, lacking coverage on some of the variable SQL clauses in the `store`. I intend to add that but to make it meaningful I need to create a CVR fixture with splits on each attribute. It would also be nice to use fixtures with BMD write-ins, but we don't handle those at all currently. So planning on doing that work in a follow-up and improving coverage.